=== modified file 'pcbnew/exporters/export_vrml.cpp'
--- pcbnew/exporters/export_vrml.cpp	2014-03-20 06:24:33 +0000
+++ pcbnew/exporters/export_vrml.cpp	2014-04-15 13:06:21 +0000
@@ -600,38 +600,36 @@
     if( text->IsMirrored() )
         NEGATE( size.x );
 
+    EDA_COLOR_T color = BLACK;  // not actually used, but needed by DrawGraphicText
+
     if( text->IsMultilineAllowed() )
     {
-        wxPoint pos = text->GetTextPosition();
         wxArrayString* list = wxStringSplit( text->GetText(), '\n' );
-        wxPoint offset;
-
-        offset.y = text->GetInterline();
-
-        RotatePoint( &offset, text->GetOrientation() );
-
-        for( unsigned i = 0; i<list->Count(); i++ )
+        std::vector<wxPoint> positions;
+        positions.reserve( list->Count() );
+        text->GetPositionsOfLinesOfMultilineText( positions, list->Count() );
+
+        for( unsigned ii = 0; ii < list->Count(); ii++ )
         {
-            wxString txt = list->Item( i );
-            DrawGraphicText( NULL, NULL, pos, BLACK,
-                    txt, text->GetOrientation(), size,
-                    text->GetHorizJustify(), text->GetVertJustify(),
-                    text->GetThickness(), text->IsItalic(),
-                    true,
-                    vrml_text_callback );
-            pos += offset;
+            wxString txt = list->Item( ii );
+            DrawGraphicText( NULL, NULL, positions[ii], color,
+                             txt, text->GetOrientation(), size,
+                             text->GetHorizJustify(), text->GetVertJustify(),
+                             text->GetThickness(), text->IsItalic(),
+                             true,
+                             vrml_text_callback );
         }
 
         delete (list);
     }
     else
     {
-        DrawGraphicText( NULL, NULL, text->GetTextPosition(), BLACK,
-                text->GetText(), text->GetOrientation(), size,
-                text->GetHorizJustify(), text->GetVertJustify(),
-                text->GetThickness(), text->IsItalic(),
-                true,
-                vrml_text_callback );
+        DrawGraphicText( NULL, NULL, text->GetTextPosition(), color,
+                         text->GetText(), text->GetOrientation(), size,
+                         text->GetHorizJustify(), text->GetVertJustify(),
+                         text->GetThickness(), text->IsItalic(),
+                         true,
+                         vrml_text_callback );
     }
 }
 

